backend/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// Directory to save uploaded files
+const uploadDir = "uploads"
+
+func uploadHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
we need to add test for this uploadHandler
we are testing any "business logic" only in the upload handler

1. lets start with simeple testing of if it checking correctly for content length.
2. then lets check if content length is not set, is the maxbytesreader is working
3. we will add more tests after these are added